test/torch_np/numpy_tests/core/test_numeric.py

```diff
@@ -371,32 +371,59 @@ def test_logical(self):
         assert_((t and s) is s)
         assert_((f and s) is f)
 
-    @xfailIfTorchDynamo
-    def test_bitwise_or(self):
+    def test_bitwise_or_eq(self):
         f = np.False_
         t = np.True_
-        assert_((t | t) is t)
-        assert_((f | t) is t)
-        assert_((t | f) is t)
-        assert_((f | f) is f)
+        assert_((t | t) == t)
+        assert_((f | t) == t)
+        assert_((t | f) == t)
+        assert_((f | f) == f)
 
-    @xfailIfTorchDynamo
-    def test_bitwise_and(self):
+    @xpassIfTorchDynamo
+    def test_bitwise_or_is(self):
         f = np.False_
         t = np.True_
-        assert_((t & t) is t)
-        assert_((f & t) is f)
-        assert_((t & f) is f)
-        assert_((f & f) is f)
+        # This deviates from proper python where (t | t) is t
+        assert_((t | t) is not t)
```

Comment:
This is not what I meant. This is not the "desired behaviour", but the behaviour we get from the semantics we chose. What I meant in https://github.com/pytorch/pytorch/pull/118978#discussion_r1476544399 is that this sort of 0D tests can be almost always fixed by doing
```suggestion
        assert_(bool((t | t)) is bool(t))
```

This preserves the semantics of the tests while falling back to eager for the execution of the part that differs between dynamo and eager.

Reply:
Yes, I like @lezcano  suggestion. We can force-cast the outputs so that eager and dynamo behavior are same.